idaes/core/util/tables.py

```diff
@@ -266,22 +267,47 @@ def stream_table_dataframe_to_string(stream_table, **kwargs):
         na_rep=na_rep, justify=justify, float_format=float_format, **kwargs
     )
 
+def _get_state_from_port(port,time_point):
+    """
+    Attempt to get a state-block-like object connected to a Port. If different
+    variables on the port appear to be connected to different state blocks,
+    raise an exception. This process is complicated by the fact that the object
+    may be indexed by space as well as time. Therefore, we assume that, if it
+    is, the time index comes first.
+    """
+    vlist = [v for v in port.iter_vars()]
+    states = [v.parent_block().parent_component() for v in vlist]
 
-def _get_state_from_port(port, time_point):
-    # Check port for _state_block attribute
+    if len(vlist) == 0:
+        raise AttributeError(
+            f"No block could be retrieved from Port {port.name} "
+            f"because it contains no components."
+            )
+    # Check the number of indices of the parent property block. If its indexed
+    # both in space and time, keep the second, spatial index and throw out the
+    # first, temporal index. If that ordering is changed, this method will
+    # need to be changed as well.
     try:
-        if len(port._state_block) == 1:
-            return port._state_block[0][time_point]
-        else:
-            return port._state_block[0][time_point, port._state_block[1]]
-    except AttributeError:
-        # Port was not created by IDAES add_port methods. Return exception for
-        # the user to fix.
-        raise ConfigurationError(
-            f"Port {port.name} does not have a _state_block attribute, "
-            f"thus cannot determine StateBlock to use for collecting data."
-            f" Please provide the associated StateBlock instead, or use "
-            f"the IDAES add_port methods to create the Port."
+        idx = vlist[0].parent_block().index()
+    except AttributeError as err:
+        raise AttributeError(
+                f"No block could be retrieved from Port {port.name} "
+                f"because block {vlist[0].parent_block()} has no index."
+                ) from err
+    # Assuming the time index is always first and the spatial indices are all
+    # the same
+    if isinstance(idx,tuple):
+        idx = (time_point,vlist[0].parent_block().index()[1:])
+
+    else:
+        idx = (time_point,)
+    # This method also assumes that ports with different spatial indices won't
+    # end up at the same port. Otherwise this check is insufficient.
+    if all(states[0] is s for s in states):
+        return states[0][idx]
+    raise AttributeError(
```

Comment:
I think RunTimeError probably describes this best, although AttributeError isn't too far off the mark. However, here we are saying that we couldn't find an object which met our criteria, and not that a specific attribute was missing from an object.